Civi/Crypto/CryptoJwt.php

```diff
@@ -60,7 +60,12 @@ public function encode($payload, $keyIdOrTag = 'SIGN') {
    * @throws CryptoException
    */
   public function decode($token, $keyTag = 'SIGN') {
-    $useKeyObj = version_compare(\Composer\InstalledVersions::getVersion('firebase/php-jwt'), '6', '>=');
+    // Version 6.x+ has 2 parameters, earlier versions had 3
+    $reflection = new \ReflectionMethod('Firebase\JWT\JWT::decode');
+    $useKeyObj = ($reflection->getNumberOfParameters() === 2) ?? FALSE;
```

Comment:
Well, this conditional distinguishes Firbase JWT [v5.x](https://github.com/firebase/php-jwt/blob/v5.5.1/src/JWT.php#L83) from [v6.0](https://github.com/firebase/php-jwt/blob/v6.0.0/src/JWT.php#L85) -- but then it misinterprets [v6.6+](https://github.com/firebase/php-jwt/blob/v6.6.0/src/JWT.php#L99).

So if your build has v6.6 `firebase/php-jwt`(*as in the D9 E2E tests*), then it misfires and gives it the v5.x data:

```
Error: Cannot pass parameter 3 by reference in Civi\Crypto\CryptoJwt->decode() (line 92 of /Users/totten/bknix/build/build-0/vendor/civicrm/civicrm-core/Civi/Crypto/CryptoJwt.php)

#0 /Users/totten/bknix/build/build-0/vendor/civicrm/civicrm-core/ext/authx/Civi/Authx/CheckCredential.php(96): Civi\Crypto\CryptoJwt->decode('')
#1 [internal function]: Civi\Authx\CheckCredential->bearerJwt(Object(Civi\Authx\CheckCredentialEvent), 'civi.authx.chec...', Object(Civi\Core\UnoptimizedEventDispatcher))
```